pkg/engine/lifecycletest/refresh_test.go

```diff
@@ -129,6 +129,70 @@ func TestExternalRefresh(t *testing.T) {
 	assert.True(t, snap.Resources[1].External)
 }
 
+// External resources should only have their outputs diffed during a refresh, in
+// line with the "legacy" implementation. Consequently Diff should not be called
+// for them. This test checks that case.
+func TestExternalRefreshDoesNotCallDiff(t *testing.T) {
+	t.Parallel()
+
+	diffCalled := false
+	loaders := []*deploytest.ProviderLoader{
+		deploytest.NewProviderLoader("pkgA", semver.MustParse("1.0.0"), func() (plugin.Provider, error) {
+			return &deploytest.Provider{
+				ReadF: func(
+					urn resource.URN,
+					id resource.ID,
+					inputs resource.PropertyMap,
+					state resource.PropertyMap,
+				) (plugin.ReadResult, resource.Status, error) {
+					return plugin.ReadResult{
+						Outputs: resource.PropertyMap{},
```

Comment:
Can we change the props here between the first `up` and `refresh` so the display snapshot shows we did look at these outputs.